### PR TITLE
Migrate old BVM values used in test cases

### LINKED ISSUE #22075: Migrate unit tests to JVM Runtime values
We still convert JBallerina values to BVM values before unit test does it's assertions.

We should get rid of the BVM values completely.

(Related to #17443)

### LINKED ISSUE #30443: Enable/remove org.ballerinalang.test.serializer.json tests
**Description:**
$title. Currently disabled. 

The associated code is also still available `org.ballerinalang.core.model.util.serializer`.

### LINKED ISSUE #35026: Migrate old BVM values in jballerina unit tests
**Description:**
$subject

**Describe your task(s)**
Continuation of task #22075

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject
Fixes #22075 
Fixes #35026 
Fixes #30443 

## Approach
Using Jvm Values for test arguments and return values. 
The sub-tasks that are done in this PR can be described as below. 
1. In `BRunUtils` class 
- Removed old APIs that used BValues
- Removed BVM to JVM value mapping for function arguments & return values
- Did some refactoring

2. Replacing BVM classes with runtime classes 
- Values -> runtime values (Eg: BMap, BError, etc.)
- Types -> runtime types (Eg: BMapType, BXmlType etc.)
- TypeTags -> runtime Typetags (needed tag name changes)
- Exceptions -> runtime exceptions

3. Major Changes in Test Files
- Tuple returning test cases. -> BArray returns
- Assertion of Simple basic values without wrappers - float(double), int(long), decimal (new DecimalValue())
- BMap access -> change the key to BString values
- Use of `getType()` API to determine type from object
- `toString()` of values - BError, BMaps, BArray
- Removal of old JSON serializer tests

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java [hunk 1]
```diff
@@ -1452,11 +416,10 @@ private static void runOnSchedule(Class<?> initClazz, BLangIdentifier name, Sche
             final Throwable t = out.panic;
             if (t != null) {
                 if (t instanceof BLangRuntimeException) {
-                    throw new org.ballerinalang.core.util.exceptions.BLangRuntimeException(t.getMessage());
```
Comment (warunalakshitha) [inline tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java:1455]: Do we need to catch BLangRuntimeException and create new BLangRuntimeException?

File: tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Nadeeshan96) [inline tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java:2]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java (hunk 1 above)
Comment (HindujaB) [reply tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java:1455]: removed with the latest commit

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**